pkg/scheduler/framework/types.go

```diff
@@ -937,7 +937,7 @@ func calculateResource(pod *v1.Pod) (Resource, int64, int64) {
 		ContainerFn: func(requests v1.ResourceList, containerType podutil.ContainerType) {
```

Comment:
we already have a correct calculation of resources as a return value of this. Would it be better to eliminate the need for an additional ContainerFn here?

Reply:
Since scheduler has a special override logic:
https://github.com/kubernetes/kubernetes/blob/0b91def6bcbdde82206ec779791762fb6bc337af/pkg/scheduler/util/pod_resources.go#L56-L67

I think we have to keep it for now, or we need to migrate this part of logic to `resource.helper`, that should be a break change?

Reply:
I agree that it's better to keep this function just here for now.

But I would be happy to see a KEP that proposes a simpler way of calculating defaults in the resource scoring that is pod-level, as opposed to container-level, outlining pros and cons of the change and the backwards-compatibility implications.

Reply:
sorry, I am not following. Why would we need a separate KEP for this and what would be a breaking change?

I really don't like the fact that we have the similar logic in two places now. It will be much cleaner to concentrate all the logic in one place.

Introducing a new Sidecar type of container (which is not a term we use in any other place in codebase) for this one place sounds like a bad idea for the future extensibility